src/Pulsar.Client/Internal/ClientCnx.fs

```diff
@@ -123,73 +125,68 @@ type ClientCnx (broker: Broker,
 
     let tryParse (buffer: ReadOnlySequence<byte>) readerId =
         let array = buffer.ToArray()
-        if (array.Length >= 8)
-        then
+        if (array.Length <= 8)
```

Comment:
= case forgotten

Reply:
you are completely right, the problem is only the `else` 50 lines later is difficult to find. Imaginary perf microoptimisations vs readability.  :)
At least I would extract a function instead, and then do it right.
I would reverse it again for you if you like the splitting DU idea generally, because I'm in doubt now whether here the right thing will be done.

So idea was just to make C#-ly `PulsarCommand` matchable with F#, or something else?

Reply:
Yeah, the main idea is in 90% cases you don't need to find this else after 50 lines, because you want to understand the general algorithm. When rare cases are on the top it's more difficult to understand what's happening. 
Generally the split makes sense for me to extract `reader.AdvanceTo consumed` code from Ok match cases, without it there is not much use.